cram/cram_encode.c

```diff
@@ -1401,6 +1402,286 @@ static int add_read_names(cram_fd *fd, cram_container *c, cram_slice *s,
 // CRAM version >= 3.1
 #define CRAM_ge31(v) ((v) >= 0x301)
 
+// Returns the next cigar op code: one of the BAM_C* codes,
+// or -1 if no more are present.
+static inline
+int next_cigar_op(uint32_t *cigar, int *ncigar, int *skip, int *spos,
```

Comment:
Regarding "skip", it could be moved there, but the `next_cigar_op` is a general purpose function (albeit static), with skip being a parameterisation of it.  I could move it there, but then it'd need renaming to something less general.  I doubt it matters much so inclined to leave as-is for now.